diff --git a/app/assets/javascripts/pages/index.js b/app/assets/javascripts/pages/index.js
index 07aa34e..94e693c 100644
--- a/app/assets/javascripts/pages/index.js
+++ b/app/assets/javascripts/pages/index.js
@@ -8882,7 +8882,7 @@ var Editor = function() {
                 })
             }
     }
-      , M = function(t, n, s) {
+      , saveStoryData = function(t, n, s) {
         var o = !1;
         if (!i.currentStoryId || i.currentStoryId === "local")
             t.url_key && t.url_key != "local" ? i.currentStoryId = t.url_key : (o = !0,
@@ -8892,7 +8892,7 @@ var Editor = function() {
         if (o)
             $.ajax({
                 type: "POST",
-                url: r + "/stories.json",
+                url: r  + "/stories.json",
                 contentType: "application/json",
                 processData: !0,
                 data: u,
@@ -8992,7 +8992,7 @@ var Editor = function() {
         signOut: N,
         signedIn: C,
         openRegisterDialogue: S,
-        saveStoryData: M,
+        saveStoryData: saveStoryData,
         localSave: k,
         hasLocalStoryData: L,
         clearSession: c,
@@ -9071,7 +9071,7 @@ var Editor = function() {
       , T = function() {
         EditorAccount.signedIn() ? EditorAccount.startNewStory() : EditorAccount.clearSession()
     }
-      , N = function(e) {
+      , OpenStory = function(e) {
         var t = new Dialogue({
             title: e.title,
             message: e.message
@@ -9137,7 +9137,7 @@ var Editor = function() {
     }
       , C = function() {
         var e = function() {
-            var e = N({
+            var e = OpenStory({
                 title: "Open",
                 message: "Choose the story to open",
                 hasDelete: !0,
@@ -9171,7 +9171,7 @@ var Editor = function() {
         e.addButton("Cancel"),
         e.addButton("Enter", function() {
             e.close();
-            var t = N({
+            var t = OpenStory({
                 title: "Competition entry",
                 message: "Choose the story you'd like to enter in the competition",
                 mainButtonTitle: "Submit",
@@ -9402,6 +9402,7 @@ var Editor = function() {
         EditorAccount.signedIn() ? F("sign out", e, J, "Sign out") : F("sign in", e, EditorAccount.signIn, "Sign in to your account"),
         F("new", e, EditorMenu.createNew, "Start a new story"),
         EditorAccount.signedIn() && F("open", e, EditorMenu.open, "Open one of your saved stories"),
+        EditorAccount.signedIn() && F("import", e, EditorMenu.importStory, "Import a story from another instance"),
         u != o ? (F("tutorial", e, EditorMenu.loadTutorial, "Load the tutorial"),
         EditorAccount.signedIn() ? u > t && F("share", e, EditorMenu.showShareDialogue, "Share your story") : F("share", e, EditorAccount.signIn, "Sign in to share your story")) : F("restart tutorial", e, EditorMenu.loadTutorial, "Restart the tutorial"),
         u != o && F("?", e, function() {
@@ -9502,9 +9503,59 @@ var Editor = function() {
     }
       , K = function() {
         var e = EditorAccount.loadLocalStory();
-        e && (Editor.load(e),
-        z())
-    };
+        e && (Editor.load(e),z())
+        }
+      , importStory = function(){
+
+        // Copy from N
+        var t = new Dialogue({
+            title: "Import",
+            message: "Paste your story to import in JSON format.",
+            footer: "After import, your story will be added to your account. Use the <b>Open</b> button to select it from the list and edit it.",
+        })
+          , n = "Import story"
+          , r = $('<form><textarea rows=20 cols=45></textarea></form>');
+
+        t.addContent(r);
+        t.addButton("Cancel");
+        t.addButton("Import story", function() {
+
+            // Based on saveStoryData
+            var story = $(".dialogue textarea").val();
+            try { 
+              JSON.parse(story); 
+            } catch(e){
+              console.log("Invalid JSON format:",e);
+              alert("Oops, your JSON format is invalid!");
+            }
+            $.ajax({
+                type: "POST",
+                url: "/stories.json",
+                contentType: "application/json",
+                processData: !0,
+                data: story,
+                success: function(jqXHR, textStatus, errorThrown) {
+                    console.log("Successfully sent data (import)."),
+                    e=0,
+                    t.close()
+                },
+                error: function(t, n, r) {
+                    e++
+                    alert("Could not save new story. Try again ?")
+                }
+            }).done(function(e) {
+                console.log("Sending (importing) done."),
+                EditorAccount.fetchStories()
+            });
+            
+            
+          })
+
+
+
+
+        return t
+      };
     return {
         eraseAndStartNew: x,
         createNew: _,
@@ -9525,7 +9576,8 @@ var Editor = function() {
         setup: V,
         update: U,
         processSignedInTasks: X,
-        loadLocalStory: K
+        loadLocalStory: K,
+        importStory: importStory,
     }
 }()
   , Player = function() {
@@ -16646,4 +16698,4 @@ rangy.createModule("Serializer", function(e, t) {
     e.restoreSelectionFromCookie = y,
     e.saveSelectionCookie = b,
     e.getElementChecksum = u
-});
+});
\ No newline at end of file
