diff --git a/app/assets/javascripts/pages/index.js b/app/assets/javascripts/pages/index.js
index 9e37949..57e563e 100644
--- a/app/assets/javascripts/pages/index.js
+++ b/app/assets/javascripts/pages/index.js
@@ -8853,7 +8853,7 @@ var Editor = function() {
         if (C())
             if (EditorMenu.unsavedChanges()) {
                 var e = navigator.cookieEnabled ? "<p>You have been disconnected from inklewriter, perhaps because of an extended period of inactivity.</p> <p>Please sign in again to save your unsaved changes.</p>" : "<p>You have been disconnected from inklewriter because your browser has cookies disabled. Please enable cookies for this site to sign in</p>";
-                _gaq.push(["_trackEvent", "Lost Connection to Server", "Signed out"]);
+                //_gaq.push(["_trackEvent", "Lost Connection to Server", "Signed out"]);
                 var t = new Dialogue({
                     title: "Signed out",
                     message: e
@@ -8888,7 +8888,7 @@ var Editor = function() {
             t.url_key && t.url_key != "local" ? i.currentStoryId = t.url_key : (o = !0,
             i.currentStoryId = "local");
         var u = jQuery.stringifyJSON(t);
-        o ? _gaq.push(["_trackEvent", "Server Interaction", "Saving", "New story", u.length / 1e3]) : _gaq.push(["_trackEvent", "Server Interaction", "Saving", "Resave of existing story", u.length / 1e3]);
+        // o ? _gaq.push(["_trackEvent", "Server Interaction", "Saving", "New story", u.length / 1e3]) : _gaq.push(["_trackEvent", "Server Interaction", "Saving", "Resave of existing story", u.length / 1e3]);
         if (o)
             $.ajax({
                 type: "POST",
