diff --git a/app/assets/javascripts/pages/index.js b/app/assets/javascripts/pages/index.js
index d99c1f5..71f1729 100644
--- a/app/assets/javascripts/pages/index.js
+++ b/app/assets/javascripts/pages/index.js
@@ -10600,7 +10600,7 @@ var Editor = function() {
                 }]
             },
             LogicNextStep: {
-                content: ['Click where it says "Always show this option" and it will bring up an interface for adding tests, which have to either be passed - or /=not=/ be passed - for the option to appear. Try it now - and try it in *-read-* mode too!', {
+                content: ['Click where it says "Add conditions" and it will bring up an interface for adding tests, which have to either be passed - or /=not=/ be passed - for the option to appear. Try it now - and try it in *-read-* mode too!', {
                     option: "Okay, I get it",
                     linkPath: "inlineTests",
                     ifConditions: null,
