diff --git a/app/assets/javascripts/pages/index.js b/app/assets/javascripts/pages/index.js
index 0c8568a..ba35dcb 100644
--- a/app/assets/javascripts/pages/index.js
+++ b/app/assets/javascripts/pages/index.js
@@ -6421,6 +6421,13 @@ var GraphNode = function(e, t, n, r) {
     },
     a
 };
+$(window).on("resize", (a,b,c) => { 
+  $([".splash", ".dialogue"]).each( (key,element) => { 
+      $(element).css("top",  ($(window).innerHeight() - $(element).height() ) / 2 ); 
+      $(element).css("left", ( $(window).innerWidth() - $(element).width() ) / 2 );
+  })
+})
+
 Dialogue.keypressBoundFunction = null,
 activeTooltip = null;
 var ToolTip = function(e, t, n) {
@@ -16763,4 +16770,4 @@ rangy.createModule("Serializer", function(e, t) {
     e.restoreSelectionFromCookie = y,
     e.saveSelectionCookie = b,
     e.getElementChecksum = u
-});
\ No newline at end of file
+});
